File: service/app/src/test/java/io/miragon/blueprint/application/service/BookCancellationCostsServiceTest.java
package io.miragon.blueprint.application.service;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.verifyNoMoreInteractions;

import io.miragon.blueprint.application.port.outbound.BikeDealerPort;
import io.miragon.blueprint.domain.bike.OrderId;
import org.junit.jupiter.api.Test;

class BookCancellationCostsServiceTest {

    private final BikeDealerPort bikeDealer = mock(BikeDealerPort.class);
    private final BookCancellationCostsService underTest = new BookCancellationCostsService(bikeDealer);

    @Test
    void bookCostsDelegatesTheCostBookingToTheDealer() {
        // given: a placed order
        OrderId orderId = new OrderId("ORDER-900");
        // when: the costs are booked
        underTest.bookCosts(orderId);
        // then: the dealer out-port books the cancellation costs
        verify(bikeDealer).bookCancellationCosts(orderId);
        verifyNoMoreInteractions(bikeDealer);
    }
}

File: service/app/src/test/kotlin/io/miragon/blueprint/application/service/BookCancellationCostsServiceTest.kt
package io.miragon.blueprint.application.service

import io.miragon.blueprint.application.port.outbound.BikeDealerPort
import io.miragon.blueprint.domain.bike.OrderId
import io.mockk.Runs
import io.mockk.confirmVerified
import io.mockk.every
import io.mockk.just
import io.mockk.mockk
import io.mockk.verify
import org.junit.jupiter.api.Test

class BookCancellationCostsServiceTest {

    private val bikeDealer = mockk<BikeDealerPort>()
    private val underTest = BookCancellationCostsService(bikeDealer = bikeDealer)

    @Test
    fun `bookCosts delegates the cost booking to the dealer`() {
        // given: a placed order
        val orderId = OrderId("ORDER-900")
        every { bikeDealer.bookCancellationCosts(orderId) } just Runs
        // when: the costs are booked
        underTest.bookCosts(orderId)
        // then: the dealer out-port books the cancellation costs
        verify { bikeDealer.bookCancellationCosts(orderId) }
        confirmVerified(bikeDealer)
    }
}
